nautobot_golden_config/utilities/helper.py

```diff
@@ -102,3 +108,72 @@ def render_jinja_template(obj, logger, template):
         )
         logger.log_failure(error_msg)
         raise NornirNautobotException from error
+
+
+def clean_config_settings(repo_type: str, repo_count: int, repo_template: str):
+    """Custom clean for `GoldenConfigSettingFeatureForm`.
+
+    Args:
+        repo_type (str): `intended` or `backup`.
+        repo_count (int): Total number of repos.
+        repo_template (str): Template str provided by user to match repos.
+
+    Raises:
+        ValidationError: Custom Validation on form.
+    """
+    if repo_count > 1:
+        if not repo_template:
+            raise forms.ValidationError(
+                f"If you specify more than one {repo_type} repository, you must provide an {repo_type} repository matching rule template."
+            )
+    elif repo_count == 1 and repo_template:
+        raise forms.ValidationError(
+            f"If you configure only one {repo_type} repository, there is no need to specify the {repo_type} repository matching rule template."
+        )
+
+
+def get_repository_working_dir(
+    repository_obj: GitRepo,
+    repo_type: str,
+    obj: Device,
+    logger: NornirLogger,
+    global_settings: models.GoldenConfigSetting,
+) -> str:
+    """Match the Device to a repository working directory, based on the repository matching rule.
+
+    Assume that the working directory == the slug of the repo.
+
+    Args:
+        repository_record (GitRepo): Git Repo Django ORM object
+        repo_type (str): Either `intended` or `backup` repository
+        obj (Device): Django ORM Device object.
+        logger (NornirLogger): Logger object
+        global_settings (models.GoldenConfigSetting): Golden Config global settings.
+
+    Returns:
+        str: The local filesystem working directory corresponding to the repo slug.
+    """
+    # Set a default for the root directory to cover the single repo use case.
+    repository_root_directory = repository_obj.path
+
+    if repo_type == "backup":
+        repo_list = global_settings.backup_repository.all()
```

Comment:
This logic mapped up with the previous seems like it could be accomplished with `global_settings.backup_repository.objects.filter(slug=desired_repository_slug)`, or similar syntax.